pallets/price-oracle/src/types.rs

```diff
@@ -0,0 +1,215 @@
+// This file is part of Basilisk-node.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode};
+use frame_support::sp_runtime::traits::{CheckedDiv, Zero};
+use frame_support::sp_runtime::RuntimeDebug;
+pub use primitives::{Balance, Price};
+use sp_std::iter::Sum;
+use sp_std::ops::{Add, Index, IndexMut, Mul};
+use sp_std::prelude::*;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(RuntimeDebug, Encode, Decode, Copy, Clone, PartialEq, Eq, Default)]
+pub struct PriceEntry {
+	pub price: Price,
+	pub amount: Balance,
+	pub liq_amount: Balance,
```

Comment:
I would prefer complete words in variable names when possible (as liq could be ambiguous)